File: 120/triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int N = triangle.size();
        int[] table = new int[N];
        List<Integer> list0 = triangle.get(N-1);
        for(int k=0; k<list0.size(); k++){
            table[k] = list0.get(k);
        }
        for(int i=N-2; i>=0; i--) {
            List<Integer> list = triangle.get(i);
            for(int k=0; k<list.size(); k++) {
                table[k] = min(table[k],table[k+1]) + list.get(k);
            }
        }
        return table[0];
    }
    int min(int a, int b) {
        if(a<b) return a;
        else return b;
    }
}

File: 120/triangle.c
//倒過來的方法，從下到上來找到結果
int min(int a, int b){
    if(a<b) return a;
    return b;
}
int minimumTotal(int** triangle, int triangleSize, int* triangleColSize){
    int M = triangleSize;
    int table[M][M];
    for(int j=0; j<M; j++){
        table[M-1][j] = triangle[M-1][j];
    } //先把底層的值放好

    for(int i=M-2; i>=0; i--){
        for(int j=0; j<triangleColSize[i]; j++){
            table[i][j] = min(table[i+1][j], table[i+1][j+1]) + triangle[i][j];
        }
    }
    return table[0][0];
}
